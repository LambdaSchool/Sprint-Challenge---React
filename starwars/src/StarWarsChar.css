.CardBody {
    display: flex;
    flex-direction: column;
}

.CardText {
    align-self: flex-start
}