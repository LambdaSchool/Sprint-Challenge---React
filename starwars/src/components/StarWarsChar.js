import React from "react";
import styled from "styled-components";

const StarWarsChars = styled.div`
  width: 40%;
  height: 300px;
  margin: 50px;
  color: white;
  background-color: #B6450D;
  border: 5px solid grey;
  border-radius:10%;
`;
const UL = styled.ul`
  list-style-type: none;
  padding: 0;
  padding-left: 40px;
`;
const H1 = styled.h1`
  font-size: 1.8rem;
  margin-top: 100px;
  
`;
const LI = styled.li``

function StarWarsChar(props) {
  return (
    <StarWarsChars>
      <UL>
        <li>
          <H1>{props.char.name}</H1>
        </li>
        <LI>Birth Year: {props.char.birth_year}</LI>
        <LI>Eye Color: {props.char.eye_color}</LI>
        <LI>Height: {props.char.height}</LI>
        <LI>Mass: {props.char.mass}</LI>
      </UL>
    </StarWarsChars>
  );
}

export default StarWarsChar;
