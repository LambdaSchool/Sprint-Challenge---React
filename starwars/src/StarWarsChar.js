import React from 'react';
import './StarWarsChar.css';
import './starWarsPersons.js'
import {
    Card,
    CardText,
    CardBody,
    CardTitle,
    CardSubtitle,
    Button
  } from "reactstrap";


  //we are not setting state here because we are setting state in the App. There is no need to
  //set state here. So we will just declare a const here. 
  const StarWarsChar = ({starwarsChar}) => {
      return (
        <div className = "Star-Wars-Card">
            <Card>
                <CardBody>
                <CardTitle>({starwarsChar.name})</CardTitle>
                <CardSubtitle>{starwarsChar.height}</CardSubtitle>
                <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
                <Button>Button</Button>
                </CardBody>
            </Card>
        </div>
      );
  };

  export default StarWarsChar;