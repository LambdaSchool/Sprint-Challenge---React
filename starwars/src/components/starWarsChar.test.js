import React from 'react';
import renderer from 'react-test-renderer';
import StarWarsChar from "../components/StarWarsChar";

describe('<App />', () => {
    it('should match snapshot', () => {
      let props = {
          name: 'Manju'
      }  
      const tree = renderer.create(<StarWarsChar char={props}/>).toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  })