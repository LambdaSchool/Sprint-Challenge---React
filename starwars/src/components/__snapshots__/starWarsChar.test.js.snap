// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="sc-bdVaJa bbrPbh"
>
  <ul
    className="sc-bwzfXH jRhgJK"
  >
    <li>
      <h1
        className="sc-htpNat gtBTPD"
      >
        Manju
      </h1>
    </li>
    <li
      className="sc-bxivhb bUZGRr"
    >
      Birth Year: 
    </li>
    <li
      className="sc-bxivhb bUZGRr"
    >
      Eye Color: 
    </li>
    <li
      className="sc-bxivhb bUZGRr"
    >
      Height: 
    </li>
    <li
      className="sc-bxivhb bUZGRr"
    >
      Mass: 
    </li>
  </ul>
</div>
`;
